ynr/apps/bulk_adding/views/sopns.py

```diff
@@ -224,10 +224,13 @@ def form_valid(self, context):
                     data["source"],
                 )
 
+            # ballot has changed so we should remove any out of date suggestions
+            ballot = Ballot.objects.get(
+                ballot_paper_id=context["ballot"].ballot_paper_id
+            )
+            ballot.delete_outdated_suggested_locks()
```

Comment:
This adds two additional queries - I think one is for the `filter` within this method, and one for the `delete()`. If we wanted to keep this down, we could have this delete happen out of band, e.g. in a regular cron job?